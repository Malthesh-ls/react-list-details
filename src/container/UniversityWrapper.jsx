import { useEffect, useState } from 'react';
import { getLocalData, setLocalData } from '../config/config';
import { ReadUniversityService } from '../services/services';
import { mapUniversityResponse } from '../utils/mappers';
import UniversityView from '../components/UniversityView';

const UniversityWrapper = () => {
  const [data, setData] = useState(null);
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(error === null);

  const onLoad = async () => {
    const { data: respData, error: apiError } = await ReadUniversityService();
    if (respData) {
      setData(mapUniversityResponse(respData));
      setLocalData(mapUniversityResponse(respData));
    } else if (apiError) {
      setData(getLocalData());
    }
    setError(apiError)
    setLoading(apiError ? !apiError : apiError);
  }

  useEffect(() => {
    onLoad();
  }, []);

  return (
    <div className='section'>
      {loading
        ? <div>Loading...</div>
        :
        <UniversityView
          data={data}
          setData={setData}
          error={error && data === null} />
      }
    </div >
  )
}

export default UniversityWrapper
