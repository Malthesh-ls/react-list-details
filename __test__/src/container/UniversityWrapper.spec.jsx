import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import UniversityWrapper from '../../../src/container/UniversityWrapper';

describe('UniversityWrapper Tests', () => {
  it('should match snapshot', () => {
    const snapshot = renderer
      .create(<UniversityWrapper />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('should display the Loading text', async () => {
    const { getByText } = render(
      <UniversityWrapper />
    );
    const text = await getByText("Loading...");
    expect(text).toBeInTheDocument();
  });
});
