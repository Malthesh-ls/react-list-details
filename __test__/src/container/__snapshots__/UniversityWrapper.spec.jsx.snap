// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniversityWrapper Tests should match snapshot 1`] = `
<div
  className="section"
>
  <div>
    Loading...
  </div>
</div>
`;
